### DISCUSSION TITLE
Getting API for data

### DISCUSSION THREAD

**Discussion by akpost21** [cu-esiil-edu/2024-data-short-course#57] [General]
Hello, 
Is there a way to create an API address for a new station besides manually writing it out (which seems to be what the API instructions indicate)? Perhaps a way to get a link after requesting the data? Thank you!
Alison

**Comment by nquarder** [comment]
Hey @akpost21 good question. I think the way I would do this would be to copy the API call that we created for the station in Boulder and then paste it into a new cell and make adjustments based on the details of the station that you'd like to look at (`station=`, `startDate=` etc.). 

For example, I'm helping @oye-bobs in [this discussion](https://github.com/cu-esiil-edu/2024-data-short-course/discussions/56) do something similar for a location in Nigeria. 

In this case the link *is* the request for the data. After you (or somebody!) writes it out once manually then you can save that like a script and make changes to it based on what you're wanting to do. 

TBH I don't think I ever wrote the entire thing out manually but instead copy/pasted one of the examples at the bottom of the instructions (see under [Example Requests](https://www.ncei.noaa.gov/support/access-data-service-api-user-documentation)) and then modified it for this example.

**Comment by akpost21** [comment]
Hi @nquarder thanks for the response! That's exactly what I ended up doing- copying and editing. Good to know that's the best method. I wasn't sure if there was another way to automatically generate the API address using the NCEI platform. Thanks for your help!

**Reply by nquarder** [reply]
Good stuff @akpost21 ! What station/city are you looking at? 🕵️

**Reply by akpost21** [reply]
I'm doing Flagstaff, AZ (since I lived there and know the weather is bonkers). @nquarder how do we submit the homework assignment?
